tests/benchmarks/timing_info/src/timing_info.h

```diff
@@ -16,23 +16,58 @@
 #define DECLARE_VAR(profile, name) \
 	u64_t total_##profile##_##name##_time;
 
-/* NRF RTC TIMER runs ar very slow rate (32KHz), So in order to measure
- * Kernel starts a dedicated timer to measure kernel stats.
+/* Stack size for all the threads created in this benchmark */
+#define STACK_SIZE 1024
+
+/******************************************************************************/
+/*
+ * Note on: SUBTRACT_CLOCK_CYCLES
+ * If we are using x86 based controller we tend to read the tsc value which is
+ * always incrementing i.e count up counter.
+ * If we are using the ARM based controllers the systick is a
+ * count down counter.
+ * If we are using nrf SOC, we are using external timer which always increments
+ * ie count up counter.
+ * Hence to calculate the cycles taken up by the code we need to adjust the
+ * values accordingly.
+ *
+ * NOTE: Needed only when reading value from end of swap operation
  */
-#if defined(CONFIG_NRF_RTC_TIMER)
-#define NANOSECS_PER_SEC 1000000000
-#define CYCLES_PER_SEC (16000000/(1 << NRF_TIMER2->PRESCALER))
+#ifdef CONFIG_NRF_RTC_TIMER
+
 /* To get current count of timer, first 1 need to be written into
  * Capture Register and Current Count will be copied into corresponding
  * current count register.
  */
-#define TIMING_INFO_PRE_READ() (NRF_TIMER2->TASKS_CAPTURE[0] = 1)
-#define TIMING_INFO_OS_GET_TIME()  (NRF_TIMER2->CC[0])
-#define TIMING_INFO_GET_TIMER_VALUE()	TIMING_INFO_GET_CURRENT_TIME()
+#define TIMING_INFO_PRE_READ()        (NRF_TIMER2->TASKS_CAPTURE[0] = 1)
```

Comment:
We really need some arch-independent APIs for getting highest-resolution timing information instead of ifdefing our test cases like this